src/container.js

```diff
@@ -1,67 +1,161 @@
 import assert from 'assert';
-import azure from 'fast-azure-storage';
+import _debug from 'debug';
+const debug = _debug('azure-blob-storage:container');
+import {rethrowDebug} from './utils';
+import {Blob, BlockBlob, AppendBlob, DataBlockBlob} from './blob';
 
-class Account {
-  constructor(opts) {
-    assert(typeof opts.accountId === 'string');
-    if (opts.accessKey) {
-      assert(typeof opts.accessKey === 'string');
-    }
-    this.accountId = opts.accountId;
-    this.accessKey = opts.accessKey || undefined;
-    this.blobsvc = new azure.Blob({
-      accountId: opts.accountId, 
-      accessKey: opts.accessKey,
-    });
+class Container {
+
+  constructor(options) {
+    options = options || {};
+    assert(typeof options.name === 'string', 'The name of the container, `options.name`, must be specified.');
+
+    this.blobService = options.blobService;
+    this.name = options.name;
+    this.eTag = options.eTag;
```

Comment:
like blob, we should only be setting this value based on the result of an API call.